dali/python/nvidia/dali/worker.py

```diff
@@ -0,0 +1,233 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+import traceback
+import os
+import socket
+from multiprocessing import reduction
+from nvidia.dali.shared_batch import SharedMemChunk, write_batch
+from nvidia.dali.messages import CompletedTasks
+
+
+class _ProcessedTasks:
+    """Internal worker message send to disptacher with completed tasks where it is
+    serialized and dispatched to the pool"""
+
+    def __init__(self, scheduled, mem_chunk=None, data_batch=None, exception=None,
+                 traceback_str=None):
+        self.context_i = scheduled.context_i
+        self.batch_i = scheduled.batch_i
+        self.mem_chunk = mem_chunk
+        self.data_batch = data_batch
+        self.exception = exception
+        self.traceback_str = traceback_str
+
+    @classmethod
+    def done(cls, scheduled, mem_chunk, data_batch):
+        return cls(scheduled, mem_chunk, data_batch)
+
+    @classmethod
+    def failed(cls, scheduled, exception, traceback_str=None):
+        return cls(scheduled, exception=exception, traceback_str=traceback_str)
+
+    def is_failed(self):
+        return self.exception is not None
+
+
+class SharedBatchesDispatcher:
```

Comment:
This name is confusing. The entity that issues processing requests to the workers is a dispatcher. This doesn't sound like one - from what I understand, there's one in each worker. This sounds more like `SharedBatchTransmitter` or `SharedBatchRelay`.

Reply:
added the function to the class, but I'm not sure if it wasn't better as a function outside.